### PR TITLE
Remove dependency from DDCMS/interface/DDShapes.h

### PR DESCRIPTION
**Pull request by namapane**
#### PR description:
Addressing #31569, replacing #31578 as discussed therein.

Should also replace the former in #31603, cf: https://github.com/cms-sw/cmssw/pull/31603#issuecomment-699986706 .

#### PR validation:

Ran MF regression over all MF geometries for all nominal currents and eras, for builders from DDD and DD4hep, from xml and from DB

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by silviodonato** [comment]
Kind reminder @cms-sw/reconstruction-l2 (this PR is needed for #31581 and #31652)

File: MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc [hunk 1]
```diff
@@ -41,25 +46,66 @@ volumeHandle::volumeHandle(const cms::DDFilteredView &fv, bool expand2Pi, bool d
...
+
+    case DDSolidShape::ddcons: {
+      DDCons cons(solid.solid());
+      double zhalf = cons.dZ();
+      double rInMinusZ = cons.rMin1();
+      double rOutMinusZ = cons.rMax1();
+      double rInPlusZ = cons.rMin2();
+      double rOutPlusZ = cons.rMax2();
+      double startPhi = cons.startPhi();
+      double deltaPhi = cons.endPhi() - startPhi;
```
Comment (jpata) [inline MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc:81]: Any issue with phi modularity? If yes, `reco::deltaPhi` can be considered.

File: MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc [hunk 2]
```diff
@@ -14,18 +14,23 @@
...
-#include "DetectorDescription/DDCMS/interface/DDShapes.h"
-
 #include <string>
 #include <iterator>
 
 using namespace SurfaceOrientation;
 using namespace std;
 using namespace magneticfield;
 using namespace edm;
+//using namespace angle_units::operators;
```
Comment (jpata) [inline MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc:24]: can be removed?

**Review by jpata — COMMENTED** [review]
From the reco point of view I had just some rather minor code remarks. The changes seem to be about geometry, therefore I would ask @cms-sw/geometry-l2 to take a look please. Thanks!

**Review by namapane — COMMENTED** [review]
>  The changes seem to be about geometry, therefore I would ask @cms-sw/geometry-l2 to take a look please.

While a look from @cms-sw/geometry-l2 (or anybody) is welcome, I'd like to point out that this is not about geometry, it is just about the internal implemention of MF with, no side effect on geometry whatsoever.
(Or maybe MF falls under the responsibility of geometry-l2; but I don't think that was ever formally defined.)

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
> From the reco point of view I had just some rather minor code remarks. The changes seem to be about geometry, therefore I would ask @cms-sw/geometry-l2 to take a look please. Thanks!

@jpata - the changes were requested by geometry and are perfectly fine :-)

**Comment by jpata** [comment]
+reconstruction

- technical, removes legacy DDShapes dependency from MagneticField/GeomBuilder
- no reco changes are observed
- additional tests have also been run for dd4hep by the author with no regressions

**Comment by silviodonato** [comment]
+1

**Pull request merged**